libs/core/src/lib/shellbar/shellbar.component.ts

```diff
@@ -26,45 +34,186 @@ export type ShellbarSizes = 's' | 'm' | 'l' | 'xl';
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ShellbarComponent implements AfterContentInit {
+export class ShellbarComponent implements AfterContentInit, AfterViewInit, OnInit, OnChanges {
     /** Size of Shellbar component 's' | 'm' | 'l' | 'xl' */
     @Input()
     size: ShellbarSizes = 'm';
 
+    /** It makes shellbar responsive */
+    @Input()
+    responsive = false;
+
+    /** Custom width can be given to 's' , 'm' , 'l' and 'xl' sizes.  */
+    @Input()
+    breakPoints = {
+        s: 320,
+        m: 720,
+        l: 1024,
+        xl: 1400
+    };
+
     /**
      * Whether the Shellbar is used with Side Navigation
      * When set to true, the responsive paddings are not applied
      */
     @Input()
     sideNav = false;
 
+    /** @hidden */
+    private _subscriptions: Subscription = new Subscription();
+
     /** @hidden */
     @ContentChild(ComboboxComponent, { static: false })
     comboboxComponent: ComboboxComponent;
 
+    /** @hidden */
+    @ContentChild(ShellbarActionsComponent)
+    _shellbarActionsComponent: ShellbarActionsComponent;
+
+    /** @hidden */
+    @ContentChild(SelectComponent, { static: false })
+    _selectComponent: SelectComponent;
+
     /** @hidden */
     @ContentChildren(forwardRef(() => ButtonComponent))
-    buttons: QueryList<ButtonComponent>;
+    _buttons: QueryList<ButtonComponent>;
+
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef, private _cdr: ChangeDetectorRef) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._attachResizeListener();
+    }
+
+    /** @hidden */
+    ngOnChanges(): void {
+        this._handleSizeChange();
```

Comment:
We should check that `size` property present in `changes` otherwise we should not perform calculations in matter of optimization.